src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -26,13 +27,13 @@ public interface MachineService {
      * @param machines - list of machines to be filled with value
      * @param value    - any object of machine sub class
      */
-    void fill(List<Object> machines, Object value);
+    void fill(List<? super Machine> machines, Machine value);
 
     /**
      * Call the method doWork() from each machine.
      * This method should be able to accept a list of bulldozers as well as list of trucks.
      *
      * @param machines - the list of machines
      */
-    void startWorking(List<Object> machines);
+    void startWorking(List<? extends Machine> machines);
```

Comment:
```suggestion
    void startWorking(List<? extends T> machines);
```

Reply:
Did I understand correctly, that we changed Machine to T everywhere, because the interface itself is parameterized with <T extends Machine>, so in future when we need to make changes we just change <T extends Machine> to <T extends SomethingElse>, instead of changing Machine to SomethingElse in each method?

Reply:
yes

Reply:
не зовсім. Якщо ми параметризуємо класс MachineService якимось класом T, то метод startWorking зможе працювати з обїектами класу T або її наслідників, а не з наслідниками класу Machine